### DISCUSSION TITLE
Apparent high CPU/memory footprint for Vertico when opening a Golang project

### DISCUSSION THREAD

**Discussion by magandrez** [minad/vertico#493] [Q&A] [answered]
Hi,

I have detected a high usage of CPU and memory usage when I am normally browsing a project. My setup is as follows:

`GNU Emacs 29.3 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.41, cairo version 1.18.0) of 2024-05-15, modified by Debian`

The high CPU and memory usage I detected it when working on a Golang project. Regarding this, I use `lsp` together with `gopls` (latest version). My complete emacs configuration (including vertico and orderless configs) can be found [here](https://github.com/magandrez/.emacs.d/blob/master/config.org).

Below, I include screens of the CPU + memory profile after opening a Golang project, along with a screenshot from htop

![screenshot_20240612_084252](https://github.com/minad/vertico/assets/743549/d33dfb6e-5874-4326-ab92-bb7c0b1c97bb)

![screenshot_20240612_084423](https://github.com/minad/vertico/assets/743549/d7044682-372b-4aa1-bb71-4c510c5f39b4)

Is this a normal allocation of resources for vertico? Do you see any potential issue in the vertico configuration working with LSP that might generate that CPU and memory footprint?

**Comment by minad** [comment, marked as answer]
Yes, resource usage is [normal](https://github.com/oantolin/orderless/issues/175#issuecomment-2160750591). Vertico and Orderless have high usage during completion in the minibuffer, since completion candidates are computed continuously. But this doesn't affect normal usage outside the minibuffer and also doesn't affect latency, since the computation is interrupted by input. If Vertico is responsive during completion, then everything is okay. For comparison, is Vertico responsive in M-x?  There you should also have thousands of candidates. Latency may go up if the completion candidate strings are long, or if there are excessively many candidates (10k+ to 50k+ depending on your system).
